plaid/plaid.go

```diff
@@ -30,39 +29,38 @@ type Client struct {
 	httpClient  *http.Client
 }
 
+type ClientOptions struct {
+	ClientID    string
+	Secret      string
+	PublicKey   string
+	Environment PlaidEnvironment
+	HttpClient  *http.Client
+}
+
 // NewClient instantiates a Client associated with a client id, secret and environment.
-func NewClient(clientID, secret, publicKey string, environment PlaidEnvironment) *Client {
-	if !environment.IsValid() {
-		fmt.Println(errors.New("Invalid environment specified: " + string(environment)))
-		return nil
+func NewClient(options ClientOptions) (client *Client, err error) {
+	if !options.Environment.Valid() {
+		return client, errors.New("Invalid environment specified: " + string(options.Environment))
```

Comment:
nit: can we return `nil, errors.New` here? It is less ambiguous that the first return argument is `nil`.

Reply:
⚡️